### ISSUE TITLE
Optional Marathon app spec attributes causing fatal error

### ISSUE THREAD

**Issue by mhausenblas** [gambol99/go-marathon#144]
When I'm [trying to ](https://github.com/mhausenblas/dploy/blob/master/src/github.com/mhausenblas/dploy/dploy/util.go#L121) read following Marathon app spec from a file and then launch it:

	"docker": {
		"image": "python:3",
		"network": "BRIDGE",
		"portMappings": [
			{
				"containerPort": 8080,
				"hostPort": 0
			}
		]
	}

then I get a:

	FATA[0000] Failed to create application null. Error: Marathon API error: Object is not valid (attribute '': )` 

I need to add: `"servicePort": 0, "protocol": "tcp"` there to make it work. This shouldn't be necessary since the latter two attributes are not required.

I'm using Marathon version  1.1.1 on DC/OS 1.7.

**Comment by timoreimann** [comment]
Thanks for the details. It seems like the request goes out to Marathon but won't be accepted, possibly because go-marathon isn't doing serialization correctly.

Will look into this, surely at the latest next week.

**Comment by mhausenblas** [comment]
Much appreciated, no hurries and KUTGW— I ❤️ your lib!

**Comment by timoreimann** [comment]
@mhausenblas glad to hear, thanks. 🌟 

I seemed to have identified the problem: [PortMapping.Protocol](https://github.com/gambol99/go-marathon/blob/93fc3db660ffd86fa7fece3e0f9d33b0d0b37494/docker.go#L36) is missing the `omitempty` tag, causing an empty string to be encoded when not explicitly setting the field otherwise. I'll put together a PR next.

Side note: I noticed that the error message returned by Marathon 0.15.2 was more clear:

```requirement failed: protocol can only be 'tcp' or 'udp' ()```

Maybe there's a chance to get back the old description again?
